graph-builder/src/release.rs

```diff
@@ -17,9 +17,9 @@ use semver::Version;
 use std::collections::HashMap;
 use std::fmt;
 
-#[derive(Debug, Deserialize, Clone)]
+#[derive(Debug, Deserialize, Clone, PartialEq)]
 pub struct Metadata {
-    kind: MetadataKind,
+    pub kind: MetadataKind,
```

Comment:
I'd prefer to keep this private. Nobody should be looking at this field.

Reply:
The issue here is that we can't create an instance of it manually if it's private:

```
error[E0451]: field `kind` of struct `net::quay_io::graph_builder::release::Metadata` is private                                             
  --> graph-builder/tests/net/quay_io/mod.rs:36:17                                                                                           
   |                                                                                                                                         
36 |                 kind: V0,                                                                                                               
   |                 ^^^^^^^^ field `kind` is private
```

What would be the simplest way to solve this? E.g., should we make a `new()` method for constructing one?

Reply:
We should just be able to implement the Default trait.

Reply:
That's what I was afraid of. Oh well, it's not the end of the world to make that public.

Reply:
I implemented my previously mentioned idea with the `new(..)` method. PTAL at https://github.com/openshift/cincinnati/pull/11/commits/4010c8cdf32cedc1e75f62788b4c38a35179cc7e#diff-f0669f981e48dab0c10160890afd503bR34. I don't think it's actually much worse than the `default()` method.

Reply:
Discussed this OOB and concluded that consumers of these objects need to be able to read this field too, so making this public is a requirement anyway.